docs/DeveloperGuide.md

```diff
@@ -604,42 +604,230 @@ testers are expected to do more *exploratory* testing.
 
-   1. Test case: `delete 1`<br>
-      Expected: First contact is deleted from the list. Details of the deleted contact shown in the status message. Timestamp in the status bar is updated.
+    2. Test case: `delete 0`<br>
+       Expected: No person is deleted. Error details shown in the status message.
 
-   1. Test case: `delete 0`<br>
-      Expected: No person is deleted. Error details shown in the status message. Status bar remains the same.
+    3. Other incorrect delete commands to try: `delete`, `delete x` (where `x` is any value), `delete n/g` (where `g` is a name that is not in KonTActs)<br>
+       Expected: Similar to previous.
+   
+### Adding a person
 
-   1. Other incorrect delete commands to try: `delete`, `delete x`, `...` (where x is larger than the list size)<br>
+1. Add a person by to KonTActs
+
+   1. Test case: `add name/John Doe email/johndoe@test.com telegram/@johndoe github/johndoe` <br>
+      Expected: Contact with name `John Doe` is added to the list. Details of the added contact shown in the status message.
+   
+   2. Test case: `add name/John Do3 email/johndo3@test.com telegram/johndoe github/john#$doe`
+      Expected: No person is added. Error details shown in the status message.
+   
+   3. Other incorrect add commands to try: `add`, `add ...` (where `...` is any pair of prefix and value)<br>
       Expected: Similar to previous.
 
-1. _{ more test cases …​ }_
+### Editing a contact
+1. Editing a contact field that is shown in the list
+   1. Test case: `edit 1 name/John Tan`
+      Expected: Contact shown at index 1 has name edited to `John Tan`. Details shown in status message.
+   
+   2. Test case: `edit 0 name/John Tan`
+      Expected: No contact is edited. Details of error shown in status message.
+
+   3. Other incorrect edit commands to try: `edit -1 name/John Tan`, `edit x name/John Tan` (where `x` is a number that is not in the shown list)
+      Expected: Similar to previous
+
+### Adding grade to a person
+1. Add grades to a person
+   1. Prerequisites: 
+      1. `assignment.json` in `data` with the following content:
+        ```json
+        {
+          "assignments" : [
+            {
+              "name": "Ex01",
+              "maxScore": 5
+            },{
+              "name": "Ex02",
+              "maxScore": 5
+            },{
+              "name": "Ex03",
+              "maxScore": 5
+            },{
+              "name": "Ex04",
+              "maxScore": 5
+            }
+          ]
+        }
+        ```
+        
+   2. Test case: `addgrade n/John Doe a/Ex04 s/5` <br>
+      Expected: Assignment `Ex04` with score `5` is added to `John Doe`. Details of the added grade shown in the status message.
+   
+   3. Test case: `addgrade/John Doe a/Ex05 s/5` <br>
+      Expected: No grade is added to John Doe. Details of the error message is shown in the status message.
+   
+   4. Other incorrect add grade commands to try: `addgrade`, `addgrade s/x` (where `x` is not a number)<br>
```

Comment:
Maybe could paraphrase the part in brackets to : where 'x' is not a number or 'x' is bigger than the max value of the assignment.